tests/core/test_signup_methods.py

```diff
@@ -50,7 +50,9 @@ def test_partially_conflicting_shift_results_in_invalid_signup_form(
         user=volunteer,
     ).form.submit(name="signup_choice", value="sign_up")
     assert "You are already confirmed for other shifts at this time" in response
-    response.form["individual_start_time_1"] = "18:42"
+    response.form[
+        "individual_start_time_1"
+    ] = "19:42"  # move start to after the conflicting event ended
```

Comment:
Now I don't now why this test was working before.
The end date ist 20:30 in the default timezone and then shifted one hour to the past, making it 19:30, so 18:42 shouldn't have worked.
I really think we should understand what is going on before merging.

Reply:
So I think I figured out what was going on. For the scenario:

There is a shift we are participating in, called the conflicting shift (cause it is meant to conflict with the signup attempted in the test). It had the same start and end times as `event.shifts.first()` (the target shift), but the end was shifted to one hour earlier up in the test in line 46.

The target shift ends at 20:30 in the current timezone, which for the tests is `Europe/Berlin`. Now the event is in June, so we need to consider DST. The correct end is therefore `20:30+02:00` or `18:30+00:00` for the main shift and `17:30+00:00` for the conflicting shift.Now date inputs in t

Now the time we are comparing against is the individual start time input from the Signup Form, a SplitDateTimeField.
I reverted back to using `'18:42'` and debugged using django 3 and django 4.

In Django 4:

The `'18:42'` gets parsed and put into a datetime with the expected DST-offset (the date is in June!) of `+02:00`, making the individual start time `16:42+00:00`, which of course means the overlap is detected an the test fails. Changing it to 19:42 resolves the flaky test.

![image](https://user-images.githubusercontent.com/5970076/156366936-2a1a112e-989e-401e-af90-1d9df2bb93f5.png)

In Django 3:

Using the 3.2.12 version however, our `'18:42'` is put into a datetime without DST, so `18:42+01:00`. I can only imagine this is because right now it's winter and DST is not in effect.

`17:42+00:00` is indeed after the conflicting shifts end date, so the test couldn't fail by claiming a conflict.

![image](https://user-images.githubusercontent.com/5970076/156367163-672cd554-533a-4d86-bb6d-35b230c24ad8.png)

Sadly, I couldn't find something in the release notes about this, but I think this PR does correct the test, and django 4.0.3 has the expected behavior in SplitDateTimeField.

Reply:
good catch!
I think this is covered in the release note with the switch from pytz to zoneinfo which just treats this differently

Reply:
Can you link me there or share a screenshot?